source/ui/CLS/CLSBeamlineStatusView.cpp

```diff
@@ -72,7 +75,7 @@ void CLSBeamlineStatusView::setBeamlineStatus(CLSBeamlineStatus *newStatus)
 
 void CLSBeamlineStatusView::setSelectedComponent(AMControl *newControl)
 {
-	if (selectedComponent_ != newControl && beamlineStatus_->components().contains(newControl)) {
+	if (selectedComponent_ != newControl && (beamlineStatus_->components().contains(newControl) || beamlineStatus_ == newControl)) {
```

Comment:
I added a button in the BeamlineStatusView to show the overall beam status (whether beam is ready or not).